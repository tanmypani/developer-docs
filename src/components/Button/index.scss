@import '../../assets/styles/variables.scss';

.button {
    font-size: $font-size-small;
    line-height: $line-height-button;
    letter-spacing: 0.6px;
    padding: 5px 16px;
    min-width: 75px;
    display: inline-block;
    text-align: center;
    border: 0;
    &.primary {
        color: $white;
        background: $darkblue;
        border-radius: 14px;
        outline: none;
    }
    &.secondary {
        color: $white;
        background: $gray-80;
        outline: none;
    }
}
