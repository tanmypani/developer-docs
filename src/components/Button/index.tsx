import React from 'react';
import './index.scss';

const Button = (props: { label: string; type: string }) => (
    <button data-testid="btn" className={`button ${props.type}`}>{props.label}</button>
);

export default Button;
