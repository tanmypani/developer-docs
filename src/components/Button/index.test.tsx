import React from "react"
import renderer from "react-test-renderer"
import { render } from "@testing-library/react"

import Button from "./index"

describe("button", () => {
  const testLabel = 'Test label';
  const testType = `Button`;

  it("should renders correctly", () => {
    const tree = renderer
      .create(<Button label={testLabel} type={testType} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it(`renders label correctly`, () => {
    const { getByTestId } = render(<Button label={testLabel} type={testType} />)

    const label = getByTestId("btn")

    expect(label).toBeInTheDocument()
  })
})