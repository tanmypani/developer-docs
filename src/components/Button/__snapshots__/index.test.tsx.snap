// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should renders correctly 1`] = `
<button
  className="button Button"
  data-testid="btn"
>
  Test label
</button>
`;
